Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -481,89 +429,12 @@
-    });
-#endif
+    m_webPage->textIndicatorStyleController().addSourceTextIndicatorStyle(session.uuid, range, resolvedRange);
 
     replaceContentsOfRangeInSession(session.uuid, resolvedRange, *fragment, hasAttributes ? WebCore::Editor::MatchStyle::No : WebCore::Editor::MatchStyle::Yes);
 
-    auto finalTextIndicatorUUID = WTF::UUID::createVersion4();
     auto characterRangeAfterReplace = WebCore::CharacterRange(range.location, range.length + characterCountDelta);
-
-    auto sessionRangeAfterReplace = contextRangeForSessionWithUUID(session.uuid);
-    if (!sessionRangeAfterReplace) {
-        ASSERT_NOT_REACHED();
-        return;
-    }
-
-    auto replacedRangeAfterReplace = UnifiedTextReplacementController::resolveCharacterRange(*sessionRangeAfterReplace, characterRangeAfterReplace);
-
-    auto remainingRange = makeRangeSelectingNodeContents(*document);
-    remainingRange.start.container = replacedRangeAfterReplace.endContainer();
-    remainingRange.start.offset = replacedRangeAfterReplace.endOffset();
-
-    auto remainingRangeUUID = WTF::UUID::createVersion4();
-    m_remainingRangesForSessions.append({ session.uuid, { remainingRangeUUID, remainingRange } });
-
-    m_webPage->createTextIndicatorForRange(replacedRangeAfterReplace, [finalTextIndicatorUUID, remainingRangeUUID, weakWebPage = WeakPtr { m_webPage }](std::optional<WebCore::TextIndicatorData>&& textIndicatorData) {
-        if (!weakWebPage)
-            return;
-        RefPtr protectedWebPage = weakWebPage.get();
-        if (textIndicatorData)
-            protectedWebPage->addTextIndicatorStyleForID(finalTextIndicatorUUID, { TextIndicatorStyle::Final, remainingRangeUUID }, *textIndicatorData);
-    });
-
-    bool attachedToExistingSession = false;
-    for (auto& [sessionUUID, sessionRangeVector] : m_textIndicatorCharacterRangesForSessions) {
-        if (sessionUUID == session.uuid) {
-            // sessionRangeVector.last() will always be valid because when creating
-            // the structure we always create the vector with an element.
-            auto characterRange = subCharacterRange(range, sessionRangeVector.last().second);
-#if PLATFORM(MAC)
-            sessionRangeVector.append({ sourceTextIndicatorUUID, characterRange });
-#endif
-            sessionRangeVector.append({ finalTextIndicatorUUID, characterRange });
-
-            attachedToExistingSession = true;
-            break;
-        }
-    }
-    if (!attachedToExistingSession) {
-#if PLATFORM(MAC)
-        m_textIndicatorCharacterRangesForSessions.append({ session.uuid, { { sourceTextIndicatorUUID, characterRangeAfterReplace }, { finalTextIndicatorUUID , characterRangeAfterReplace } } });
-#else
-        m_textIndicatorCharacterRangesForSessions.append({ session.uuid, { { finalTextIndicatorUUID, characterRangeAfterReplace } } });
-#endif
-    }
+    m_webPage->textIndicatorStyleController().addDestinationTextIndicatorStyle(session.uuid, characterRangeAfterReplace, resolvedRange);
```

Comment:
I think this should just call a method on WebPage that calls a method on `textIndicatorStyleController`, instead of exposing the controllers to each other